pennylane/math/multi_dispatch.py

```diff
@@ -1065,3 +1065,44 @@ def set_index(array, idx, val, like=None):
 
     array[idx] = val
     return array
+
+
+@multi_dispatch(tensor_list=[1])
+def add_index(array, idx, val, like=None):
+    """Add the value at a specified index in an array.
+    Calls ``array[idx]+=val`` and returns the updated array unless JAX.
+
+    If `idx` and `val` are TensorLike, then they must be the same length
+
+    Args:
+        array (tensor_like): array to be modified (unless JAX)
+        idx (int, tensor_like): index to modify
+        val (Union[int, float, complex, tensor_like]): value to update with
+
+    Returns:
+        a new copy of the array with the specified index updated by ``val``.
+
+    Whether the original array is modified is interface-dependent.
+
+    .. note:: TensorFlow EagerTensor does not support item assignment
+    """
+    if like == "jax":
+        from jax import numpy as jnp
+
+        # ensure array is jax array (interface may be jax because of idx or val and not array)
+        jax_array = jnp.array(array)

```

Comment:
Should this be asarray to avoid potential copy?
```suggestion
        jax_array = jnp.asarray(array)

```

Reply:
Ahh I wasn't aware that it creates a copy under the hood.